mdapi/MetricsDiscoveryHelper.cpp

```diff
@@ -0,0 +1,1142 @@
+/************************************************************************/
+MDHelper::~MDHelper()
+{
+    DebugPrint("MDHelper Destructor");
+
+    if(CloseMetricsDevice != NULL && m_MetricsDevice)
+    {
+        CloseMetricsDevice( m_MetricsDevice );
+        m_MetricsDevice = NULL;
+    }
+
+    m_Initialized = false;
+}
+
+/************************************************************************/
+/* InitMetricsDiscovery                                                 */
+/************************************************************************/
+bool MDHelper::InitMetricsDiscovery(
+    const std::string& metricSetSymbolName,
+    const std::string& metricsFileName )
+{
+    m_Initialized = false;
+
+    CloseMetricsDevice = NULL;
+    OpenMetricsDevice = NULL;
+    OpenMetricsDeviceFromFile = NULL;
+
+    TCompletionCode res = CC_OK;
+
+    if (m_APIMask & API_TYPE_IOSTREAM )
+    {
+        DebugPrint("API type must not be IOSTREAM.");
+        return false;
+    }
+
+    void* pLibrary = OpenLibrary(cMDLibFileName);
+    if (pLibrary == NULL)
+    {
+        DebugPrint("Couldn't load metrics discovery library!");
+        return false;
+    }
+
+    CloseMetricsDevice = (CloseMetricsDevice_fn)GetFunctionAddress(pLibrary, "CloseMetricsDevice");
+    if (CloseMetricsDevice == NULL)
+    {
+        DebugPrint("CloseMetricsDevice NULL, error: %d", GetLastError());
+        return false;
+    }
+
+    OpenMetricsDevice = (OpenMetricsDevice_fn)GetFunctionAddress(pLibrary, "OpenMetricsDevice");
+    if (OpenMetricsDevice == NULL)
+    {
+        DebugPrint("OpenMetricsDevice NULL, error: %d", GetLastError());
+        return false;
+    }
+
+    OpenMetricsDeviceFromFile = (OpenMetricsDeviceFromFile_fn)GetFunctionAddress(pLibrary, "OpenMetricsDeviceFromFile");
+    if (OpenMetricsDeviceFromFile == NULL)
+    {
+        DebugPrint("OpenMetricsDeviceFromFile NULL, error: %d", GetLastError());
+        return false;
+    }
+
+    if (!metricsFileName.empty())
+    {
+        res = OpenMetricsDeviceFromFile(metricsFileName.c_str(), (void*)"", &m_MetricsDevice);
+        if (res != CC_OK)
+        {
+            DebugPrint("OpenMetricsDeviceFromFile failed, res: %d", res);
+            return false;
+        }
+    }
+    else
+    {
+        res = OpenMetricsDevice(&m_MetricsDevice);
+        if (res != CC_OK)
+        {
+            DebugPrint("OpenMetricsDevice failed, res: %d", res);
+            return false;
+        }
+    }
+
+    TMetricsDeviceParams_1_0* deviceParams = m_MetricsDevice->GetParams();
+    if (NULL == deviceParams)
+    {
+        DebugPrint("DeviceParams null");
+        return false;
+    }
+
+    DebugPrint("MDAPI Headers: v%d.%d.%d, MDAPI Lib: v%d.%d.%d\n",
+        MD_API_MAJOR_NUMBER_CURRENT,
+        MD_API_MINOR_NUMBER_CURRENT,
+        MD_API_BUILD_NUMBER_CURRENT,
+        deviceParams->Version.MajorNumber,
+        deviceParams->Version.MinorNumber,
+        deviceParams->Version.BuildNumber);
+
```

Comment:
You should check MDAPI version to be at least 1.1.X (unless you use any newer features then, the version should be checked against the higher number).